src/main/java/com/endava/endabank/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,240 @@
+import com.endava.endabank.exceptions.customexceptions.ResourceNotFoundException;
+import com.endava.endabank.exceptions.customexceptions.ServiceUnavailableException;
+import com.endava.endabank.exceptions.customexceptions.UniqueConstraintViolationException;
+import com.endava.endabank.model.ForgotUserPasswordToken;
+import com.endava.endabank.model.IdentifierType;
+import com.endava.endabank.model.Permission;
+import com.endava.endabank.model.Role;
+import com.endava.endabank.model.User;
+import com.endava.endabank.configuration.MailProperties;
+import com.endava.endabank.security.utils.JwtManage;
+import com.endava.endabank.service.ForgotUserPasswordTokenService;
+import com.endava.endabank.service.IdentifierTypeService;
+import com.endava.endabank.service.RoleService;
+import com.endava.endabank.service.UserService;
+import com.endava.endabank.utils.MailService;
+import com.endava.endabank.utils.user.UserValidations;
+import com.google.common.annotations.VisibleForTesting;
+import com.sendgrid.Response;
+import lombok.AllArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.BiFunction;
+
+@Service
+@AllArgsConstructor
+public class UserServiceImpl implements UserService {
+    private UserDao userDao;
+    private ModelMapper modelMapper;
+    private IdentifierTypeService identifierTypeService;
+    private RoleService roleService;
+    private PasswordEncoder passwordEncoder;
+    private ForgotUserPasswordTokenService forgotUserPasswordTokenService;
+    private MailService mailService;
+    private MailProperties mailProperties;
+
+
+    @Override
+    @Transactional(readOnly = true)
+    public List<UserToApproveAccountDto> usersToApprove() {
+        List<User> users = userDao.findAll();
+        return users.stream().map(this::mapToUserToApproveDto).toList();
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public User findById(Integer id) {
+        return userDao.findById(id).
+                orElseThrow(() -> new UsernameNotFoundException(Strings.USER_NOT_FOUND));
+    }
+
+    @Override
+    @Transactional
+    public User save(UserRegisterDto userDto) {
+        User user = modelMapper.map(userDto, User.class);
```

Comment:
Hi, 

What if `modelMapper.map(userDto, User.class);` fails?

Reply:
hi,
Why should it fail, knowing that we have the userDto validated, and well if it fails for some unknown reason it'll go to the @controllerAdvice

Reply:
When there are differences between the classes to map or another kind of causes, something can fail and there will be an exception thrown